cloudify_cli/commands/cluster.py

```diff
@@ -131,39 +130,70 @@ def join(client,
     current members are unreachable, but is not required.
     """
     _verify_not_in_cluster(client)
-    if cluster_join_profile:
-        if not env.is_profile_exists(cluster_join_profile):
-            raise CloudifyCliError('No such profile: {0}'
-                                   .format(cluster_join_profile))
-        joined_profile = env.get_profile_context(cluster_join_profile)
-        if not joined_profile.cluster:
-            raise CloudifyCliError('Cannot join profile {0} - that profile '
-                                   'has no cluster started'
-                                   .format(cluster_join_profile))
-
+    if not env.is_profile_exists(join_profile):
+        raise CloudifyCliError('No such profile: {0}'
+                               .format(join_profile))
+    joined_profile = env.get_profile_context(join_profile)
+    if not joined_profile.cluster:
+        raise CloudifyCliError('Cannot join profile {0} - that profile '
+                               'has no cluster started'
+                               .format(join_profile))
+
+    deadline = time.time() + timeout
+    cluster_client = env.get_rest_client(
+        rest_host=joined_profile.manager_ip,
+        rest_port=joined_profile.rest_port,
+        rest_protocol=joined_profile.rest_protocol)
+    cluster_status = cluster_client.cluster.status()
+
+    encryption_key = cluster_status.encryption_key
+    join = [n.host_ip for n in cluster_client.cluster.nodes.list()]
     logger.info('Joining the Cloudify Manager cluster: {0}'
-                .format(cluster_join))
+                .format(join))
 
     client.cluster.join(config={
         'host_ip': cluster_host_ip,
         'node_name': cluster_node_name,
-        'consul_key': cluster_encryption_key,
-        'join_addrs': cluster_join
+        'encryption_key': encryption_key,
+        'join_addrs': join
     })
-    status = _wait_for_cluster_initialized(client, logger)
+    try:
+        status = _wait_for_cluster_initialized(client, logger, timeout=timeout)
```

Comment:
wouldn't it be more precise/safe if we'll pass the calculated deadline, instead of the timeout arg?